notebooks/03-query/09-functions-case.ipynb

```diff
@@ -0,0 +1,499 @@
+    "      CASE \"browser\"\n",
+    "          WHEN 'IE' THEN session_length * 2\n",
+    "          WHEN 'Chrome' THEN session_length / 2\n",
+    "          ELSE session_length\n",
+    "          END\n",
+    "          ) AS \"average_session_length_maybe\"\n",
+    "FROM \"example-koalas-conditions\"\n",
+    "WHERE TIME_IN_INTERVAL(__time,'2019-08-25T0/PT1H')\n",
+    "GROUP BY 1\n",
+    "'''\n",
+    "\n",
+    "display.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "29bb64c9-9c4c-476b-9718-51c1c35a3d05",
+   "metadata": {},
+   "source": [
+    "### Searched CASE\n",
+    "\n",
+    "The second form of `CASE` is not pinned to a specific dimension or expression, allowing for more complex conditions.\n",
+    "\n",
+    "A common use of `CASE` is to tag particular events as significant.\n",
+    "\n",
+    "With some artistic license, we can apply this to the example dataset. The SQL statement below flags some events as impressions (when someone first saw an image) and others as conversions (when someone completed the task and finished clearing the image)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e8f68a70-e629-4ba7-a28d-5ba282d8469b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql='''\n",
+    "SELECT\n",
+    "  CASE WHEN (\"event_type\" = 'GoodLoad') THEN 'yes' ELSE 'no' END AS \"isImpression\",\n",
+    "  CASE WHEN (\"event_subtype\" IS NULL AND \"event_type\" = 'LayerClear') THEN 'yes' ELSE 'no' END AS \"isConversion\",\n",
+    "  COUNT(*) AS \"Count\"\n",
+    "FROM \"example-koalas-conditions\"\n",
+    "WHERE TIME_IN_INTERVAL(__time,'2019-08-25T0/PT4H')\n",
+    "GROUP BY 1, 2\n",
+    "'''\n",
+    "\n",
+    "display.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b266e5bb-a4b8-4976-94b4-e94d8d257b44",
+   "metadata": {},
+   "source": [
+    "Combining `CASE` with `SUM` leads to an interesting pattern: the SQL below shows an approach to finding the number of conversions per image shown."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f1f45089-ae33-4181-a91f-7006defa7247",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql='''\n",
+    "SELECT\n",
+    "  REGEXP_EXTRACT(\"loaded_image\",'[^/\\\\&\\?]+\\.\\w{3,4}(?=([\\?&].*$|$))') AS \"filename\",\n",
+    "  SUM(CASE WHEN (\"event_type\" = 'GoodLoad') THEN 1 ELSE 0 END) AS \"impressions\",\n",
+    "  SUM(CASE WHEN (\"event_subtype\" IS NULL AND \"event_type\" = 'LayerClear') THEN 1 ELSE 0 END) AS \"conversions\"\n",
+    "FROM \"example-koalas-conditions\"\n",
+    "WHERE TIME_IN_INTERVAL(__time,'2019-08-25T0/PT4H')\n",
+    "GROUP BY 1\n",
+    "'''\n",
+    "display.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9a995573-8991-4f89-a36b-deabcf70cd86",
+   "metadata": {},
+   "source": [
+    "Run the following cell to return a new dimension that takes the percent cleared and turns it into a funnel stage."
```

Comment:
I only have a slight grasp on what this means -- maybe explain how to interpret a funnel stage in this context (and possibly how it can apply to ad analytics)

Reply:
Thanks - added some commentary.